params/superchain.go

```diff
@@ -48,10 +48,13 @@ func LoadOPStackChainConfig(chainID uint64) (*ChainConfig, error) {
 	}
 
 	optimism := OptimismConfig{
-		EIP1559Elasticity:        chConfig.Optimism.EIP1559Elasticity,
-		EIP1559DenominatorCanyon: uint64ptr(*chConfig.Optimism.EIP1559DenominatorCanyon),
-		EIP1559Denominator:       chConfig.Optimism.EIP1559Denominator,
+		EIP1559Elasticity:  chConfig.Optimism.EIP1559Elasticity,
+		EIP1559Denominator: chConfig.Optimism.EIP1559Denominator,
 	}
+	if chConfig.Optimism.EIP1559DenominatorCanyon != nil {
+		optimism.EIP1559DenominatorCanyon = uint64ptr(*chConfig.Optimism.EIP1559DenominatorCanyon)
+	}
```

Comment:
I had to make the loading code a little more complex to prevent problems with tests, which were mutating pointers nested inside Genesis structs and causing some nasty crosstalk.